libcaf_core/caf/span.hpp

```diff
@@ -207,7 +207,7 @@ span<std::byte> as_writable_bytes(span<T> xs) {
 /// Convenience function to make using `caf::span` more convenient without the
 /// deduction guides.
 template <class T>
-auto make_span(T& xs) -> span<std::remove_reference_t<decltype(xs[0])>> {
+auto make_span(T&& xs) -> span<std::remove_reference_t<decltype(xs[0])>> {
```

Comment:
Please don't, this was on purpose to avoid having `make_span` take the address of a temporary. Allow rvalues here can lead to very nasty and hard to debug bugs.

Reply:
Reverted back. 
Having a overload for temporary non-owning data would be beneficial though.